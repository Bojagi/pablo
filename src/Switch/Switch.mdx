---
name: Switch
route: /switch
menu: Components
---

import { Props, Playground } from 'docz'
import { Box } from '../Box/Box.tsx'
import { Switch } from './Switch.tsx'

# Switch

Form element that allows users to toggle a boolean value. This is useful for filters and boolean settings.
The `Switch` components shares an interface with the `Radio` and `Checkbox` components.

## Examples

### Single switch

<Playground>
  {() => {
    const [checked, setChecked] = React.useState(false);
    return (
      <Switch checked={checked} onChange={() => setChecked(!checked)} />
    );
  }}
</Playground>

### Switches with labels

You can pass an optional `label` prop with a `ReactNode` value to show a label next to the checkbox.

<Playground>
  {() => {
    const [checked, setChecked] = React.useState(false);
    const [checkedTwo, setCheckedTwo] = React.useState(true);
    return (
      <Box my={-4}>
        <Switch my={4} label="Spaghetti mode" checked={checked} onChange={() => setChecked(!checked)} />
        <Switch my={4} label="Pizza mode" checked={checkedTwo} onChange={() => setCheckedTwo(!checkedTwo)} />
      </Box>
    );
  }}
</Playground>

### Small switches

You can pass the `size` prop with the `small` value which renders a smaller version.

<Playground>
  {() => {
    const [checked, setChecked] = React.useState(false);
    const [checkedTwo, setCheckedTwo] = React.useState(true);
    return (
      <Box display="flex" mx={-2}>
        <Switch mx={2} size="small" label="one" checked={checked} onChange={() => setChecked(!checked)} />
        <Switch mx={2} size="small" label="two" checked={checkedTwo} onChange={() => setCheckedTwo(!checkedTwo)} />
      </Box>
    );
  }}
</Playground>

## Import

```js
import { Switch } from '@bojagi/pablo/Switch';
```

## Props
<Props of={Switch} />
