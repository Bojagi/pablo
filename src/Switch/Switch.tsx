import React from 'react';
import styled, { css } from 'styled-components';
import { getComponentStyle, transitionTransformer } from '../utils/styleHelpers/getComponentStyle';
import {
  BaseCheckable,
  BaseCheckableProps,
  CheckableBoxProps,
  CheckableHandleProps,
} from '../BaseCheckable/BaseCheckable';

const SwitchBox = styled.div<CheckableBoxProps>`
  position: relative;
  width: calc(2 * ${(props) => getComponentStyle('switch.handleSize.{size}')(props)});
  height: ${(props) => getComponentStyle('switch.handleSize.{size}')(props)};
  border-radius: calc(
    (
        ${(props) => getComponentStyle('switch.handleSize.{size}')(props)} + 2 *
          ${getComponentStyle('switch.innerPadding.{size}')} + 2 *
          ${getComponentStyle('switch.borderWidth')}px
      ) * 0.5
  );
  padding: ${getComponentStyle('switch.innerPadding.{size}')};
  background-color: ${getComponentStyle('switch.backgroundColor')};
  border: ${getComponentStyle('switch.borderWidth')}px solid
    ${getComponentStyle('switch.borderColor')};
  opacity: ${(props) => (props.disabled ? 0.3 : 1)};

  transition: ${getComponentStyle('switch.boxTransition', transitionTransformer)};

  ${(props) =>
    props.focus &&
    css`
      box-shadow: 0 0 0 ${getComponentStyle('switch.focus.outlineSize')}
        ${getComponentStyle('switch.focus.outlineColor')};
    `};
  }
`;

const SwitchHandle = styled.div<CheckableHandleProps>`
  width: ${getComponentStyle('switch.handleSize.{size}')};
  height: ${getComponentStyle('switch.handleSize.{size}')};
  transform: translateX(
    ${(props: any) => (props.checked ? getComponentStyle('switch.handleSize.{size}')(props) : 0)}
  );
  border-radius: 50%;
  transition: ${getComponentStyle('switch.handleTransition', transitionTransformer)};
  background-color: ${(props) =>
    props.checked
      ? getComponentStyle('switch.handleColorChecked')(props)
      : getComponentStyle('switch.handleColorUnchecked')(props)};
`;

export interface SwitchProps extends BaseCheckableProps {}

export const Switch = (props: SwitchProps) => (
  <BaseCheckable
    componentName="switch"
    componentType="checkbox"
    componentBox={SwitchBox}
    componentHandle={SwitchHandle}
    {...props}
  />
);
