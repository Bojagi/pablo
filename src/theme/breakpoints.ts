export type Breakpoint = 'sm' | 'md' | 'lg' | 'xl';

export const breakpointNames: Breakpoint[] = ['sm', 'md', 'lg', 'xl'];

export type BreakpointsArray = [number, number, number, number];

export type Breakpoints = BreakpointsArray &
  Record<Breakpoint, number> & {
    breakpointNames: Breakpoint[];
  };

export function createBreakpoints(
  array: BreakpointsArray,
  usedBreakpointNames: Breakpoint[] = breakpointNames
): Breakpoints {
  const newBreakpoints = ([...array] as unknown) as Breakpoints;
  usedBreakpointNames.forEach((bp, index) => {
    newBreakpoints[bp] = newBreakpoints[index];
  });
  newBreakpoints.breakpointNames = usedBreakpointNames;

  return newBreakpoints as Breakpoints;
}

export const defaultBreakpoints: BreakpointsArray = [600, 1000, 1200, 1920];

export const breakpoints = createBreakpoints(defaultBreakpoints, breakpointNames);
