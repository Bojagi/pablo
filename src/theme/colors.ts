export type ShadedColor = {
  main: string;
  dark: string;
  darkest: string;
  light: string;
  lightest: string;
  contrastText: string;
  contrastTextLight: string;
};

export type CommonColors = {
  white: string;
  black: string;
  whiteContrastText: string;
  blackContrastText: string;
};

export type TextColors = {
  main: string;
  info: string;
};

export type BorderColors = {
  main: string;
  lightest: string;
};

export interface GrayColors {
  '50': string;
  '100': string;
  '200': string;
  '300': string;
  '400': string;
  '500': string;
  '600': string;
  '700': string;
  '800': string;
  '900': string;
}

export type AllColors = ShadedColor & TextColors & BorderColors & CommonColors;

export interface Colors {
  common: CommonColors;
  gray: GrayColors;
  borders: BorderColors;
  brand: ShadedColor;
  positive: ShadedColor;
  negative: ShadedColor;
  neutral: ShadedColor;
  text: TextColors;
}

export const WHITE = '#ffffff';
export const BLACK = '#000000';

export const colors: Colors = {
  common: {
    white: WHITE,
    black: BLACK,
    whiteContrastText: BLACK,
    blackContrastText: WHITE,
  },
  gray: {
    '50': getGrayColor(0.95),
    '100': getGrayColor(0.9),
    '200': getGrayColor(0.8),
    '300': getGrayColor(0.7),
    '400': getGrayColor(0.6),
    '500': getGrayColor(0.5),
    '600': getGrayColor(0.4),
    '700': getGrayColor(0.3),
    '800': getGrayColor(0.2),
    '900': getGrayColor(0.1),
  },
  borders: {
    main: `${BLACK}${getHexByte(0.25)}`,
    lightest: `${BLACK}${getHexByte(0.1)}`,
  },
  text: {
    main: BLACK,
    info: `${BLACK}${getHexByte(0.5)}`,
  },
  brand: {
    lightest: '#EBE7FF',
    light: '#D6CEFF',
    main: '#6A50F2',
    dark: '#4129BD',
    darkest: '#281683',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  positive: {
    lightest: '#DFF6E6',
    light: '#BBECCB',
    main: '#38C765',
    dark: '#2FB85A',
    darkest: '#108B36',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  negative: {
    lightest: '#FFE7E5',
    light: '#FFCFCC',
    main: '#F21D0D',
    dark: '#CA271B',
    darkest: '#871A12',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  neutral: {
    lightest: '#FCF5DB',
    light: '#FFEB9E',
    main: '#FFDD5B',
    dark: '#EDC52A',
    darkest: '#D6AC0B',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
};

/**
 * Get hex color byte (e.g. to get transparency)
 * @param value Percentage value (between 0.0 and 1.0)
 */
function getHexByte(value: number) {
  return (Math.round((255 * value) / 16) * 16).toString(16);
}

function getGrayColor(value: number) {
  return `#${getHexByte(value)}${getHexByte(value)}${getHexByte(value)}`;
}
