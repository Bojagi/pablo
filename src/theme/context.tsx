import * as React from 'react';
import { PabloTheme, ComponentStyles } from './types';

export const pabloThemeContext = React.createContext<PabloTheme>({} as any);
export const pabloComponentStylesContext = React.createContext<ComponentStyles>({} as any);
export const rootContext = React.createContext<Document | ShadowRoot | null>(null);

export const useComponentStyleContext = () => React.useContext(pabloComponentStylesContext);

export const usePabloTheme = () => React.useContext(pabloThemeContext);
