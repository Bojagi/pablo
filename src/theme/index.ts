export * from './defaultTheme';
export * from './context';
export * from './PabloThemeProvider';
export * from './useComponentStyle';
