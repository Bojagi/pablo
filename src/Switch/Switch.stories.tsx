import React from 'react';
import { Box } from '../Box';
import { Switch, SwitchProps } from './Switch';

export default {
  title: 'Switch',
};

const ControlledSwitch = ({ checked: checkedInitial, ...props }: Omit<SwitchProps, 'onChange'>) => {
  const [checked, setChecked] = React.useState(checkedInitial);
  return <Switch checked={checked} {...props} onChange={() => setChecked(!checked)} />;
};

const baseStory = (args) => (
  <Box flex mx={-1} mb={1}>
    <ControlledSwitch mx={1} checked={true} {...args} />
    <ControlledSwitch mx={1} checked={false} {...args} />
    <ControlledSwitch mx={1} checked={true} disabled={true} {...args} />
  </Box>
);

export const Medium = baseStory.bind(null);
Medium.args = {
  size: 'medium',
};

export const Small = baseStory.bind(null);
Small.args = {
  size: 'small',
};

export const WithLabel = () => (
  <>
    <Box mb={4}>
      <ControlledSwitch checked={true} label="Hans Wurst" />
    </Box>
    <Box mb={4}>
      <ControlledSwitch size="small" checked={true} label="Hans Wurst" />
    </Box>
  </>
);
