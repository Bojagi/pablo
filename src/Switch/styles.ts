import { getSpacing } from '../styleHelpers/getSpacing';
import { Style } from '../theme/types';
import { checkableBaseStyles, CheckableBaseStyles } from '../shared/checkableBaseStyles';
import { themeVars } from '../theme/themeVars';

export interface SwitchStyles extends CheckableBaseStyles {
  handleColorChecked: Style;
  handleColorUnchecked: Style;
}

export const switchStyles: SwitchStyles = {
  ...checkableBaseStyles,
  innerPadding: {
    small: getSpacing(1),
    medium: getSpacing(1),
  },
  handleSize: {
    medium: '18px',
    small: '12px',
  },
  handleColorChecked: themeVars.colors.brand.main,
  handleColorUnchecked: themeVars.colors.gray[500],
  boxTransition: [['box-shadow', '0.3s', 'ease-in-out']],
  handleTransition: [
    ['background-color', '0.15s', 'ease-in-out'],
    ['transform', '0.15s', 'ease-in-out'],
  ],
};
