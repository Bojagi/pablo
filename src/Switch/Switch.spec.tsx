import { render, cleanup, act, fireEvent } from '@testing-library/react';
import React from 'react';
import { defaultTheme, PabloThemeProvider } from '../theme';
import { Switch } from './Switch';

let onChangeMock;

beforeEach(() => {
  onChangeMock = jest.fn();
});

afterEach(cleanup);

test('Render switch unchecked', () => {
  const { container, getByTestId } = renderComponent({ checked: false });
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-switch-handle')).toHaveStyleRule('transform', 'translateX( 0 )');
  expect(getByTestId('pbl-switch-handle')).toHaveStyleRule(
    'background-color',
    defaultTheme.colors.gray['500']
  );
  expect(getByTestId('pbl-switch-input')).not.toBeChecked();
});

test('Render switch checked', () => {
  const { container, getByTestId } = renderComponent({ checked: true });
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-switch-handle')).toHaveStyleRule('transform', 'translateX( 18px )');
  expect(getByTestId('pbl-switch-handle')).toHaveStyleRule(
    'background-color',
    defaultTheme.colors.brand.main
  );
  expect(getByTestId('pbl-switch-input')).toBeChecked();
});

test('Render switch with label', () => {
  const { getByTestId } = renderComponent({
    checked: true,
    label: (
      <>
        I am a <strong>label</strong>
      </>
    ),
  });
  expect(getByTestId('pbl-switch-label')).toBeDefined();

  // Label and input are connected via "id"
  expect(getByTestId('pbl-switch-label')).toHaveAttribute('for', 'custom-id');
  expect(getByTestId('pbl-switch-input')).toHaveAttribute('id', 'custom-id');
  expect(getByTestId('pbl-switch-label')).toMatchSnapshot();
  expect(getByTestId('pbl-switch-label')).toHaveTextContent('I am a label');
});

test('Click on switch to trigger onChange event', () => {
  const { getByTestId } = renderComponent({
    checked: false,
    value: 'checky',
  });
  expect(onChangeMock).toHaveBeenCalledTimes(0);
  expect(getByTestId('pbl-switch-input')).toBeEnabled();
  act(() => {
    fireEvent.click(getByTestId('pbl-switch-input'), { target: { value: 'checky' } });
  });
  expect(onChangeMock).toHaveBeenCalledTimes(1);
  expect(onChangeMock).toHaveBeenCalledWith('checky', expect.objectContaining({ type: 'change' }));
});

test('Disabled prop should disable input', () => {
  const { getByTestId } = renderComponent({
    checked: true,
    disabled: true,
  });
  expect(getByTestId('pbl-switch-input')).toBeDisabled();
});

test('Render medium switch', () => {
  const { getByTestId } = renderComponent({
    size: 'medium',
    label: 'medium label',
  });
  expect(getByTestId('pbl-switch')).toHaveStyleRule('width', 'calc(2 * 18px)');
  expect(getByTestId('pbl-switch')).toHaveStyleRule('height', '18px');
  expect(getByTestId('pbl-switch')).toHaveStyleRule('padding', '2px');
  expect(getByTestId('pbl-switch-handle')).toHaveStyleRule('width', '18px');
  expect(getByTestId('pbl-switch-handle')).toHaveStyleRule('height', '18px');
  // Uses subtitle typography
  expect(getByTestId('pbl-switch-label').childNodes[0]).toHaveStyleRule(
    'font-size',
    defaultTheme.typography.subtitle.fontSize
  );
});

test('Render small switch', () => {
  const { getByTestId } = renderComponent({
    size: 'small',
    label: 'small label',
  });
  expect(getByTestId('pbl-switch')).toHaveStyleRule('width', 'calc(2 * 12px)');
  expect(getByTestId('pbl-switch')).toHaveStyleRule('height', '12px');
  expect(getByTestId('pbl-switch')).toHaveStyleRule('padding', '2px');
  expect(getByTestId('pbl-switch-handle')).toHaveStyleRule('width', '12px');
  expect(getByTestId('pbl-switch-handle')).toHaveStyleRule('height', '12px');
  // Uses paragraph typography
  expect(getByTestId('pbl-switch-label').childNodes[0]).toHaveStyleRule(
    'font-size',
    defaultTheme.typography.paragraph.fontSize
  );
});

test('Use generated id when no "id" prop is set', () => {
  const { getByTestId } = renderComponent({
    id: undefined,
  });

  expect(getByTestId('pbl-switch-input')).toHaveAttribute(
    'id',
    expect.stringContaining('pbl-switch-')
  );
});

test('Render without onChange function ', () => {
  renderComponent({
    onChange: undefined,
  });
});

test('Forward onFocus and onBlur to input and show focus outline', () => {
  const onFocusMock = jest.fn();
  const onBlurMock = jest.fn();
  const { getByTestId } = renderComponent({
    onChange: undefined,
    checked: undefined,
    onFocus: onFocusMock,
    onBlur: onBlurMock,
  });

  expect(onFocusMock).toHaveBeenCalledTimes(0);
  act(() => {
    fireEvent.focus(getByTestId('pbl-switch-input'));
  });
  expect(onFocusMock).toHaveBeenCalledTimes(1);

  expect(getByTestId('pbl-switch')).toHaveStyleRule(
    'box-shadow',
    `0 0 0 3px ${defaultTheme.colors.brand.lightest}`
  );

  expect(onBlurMock).toHaveBeenCalledTimes(0);
  act(() => {
    fireEvent.blur(getByTestId('pbl-switch-input'));
  });
  expect(onBlurMock).toHaveBeenCalledTimes(1);

  expect(getByTestId('pbl-switch')).toHaveStyleRule('box-shadow', undefined);
});

function renderComponent(props) {
  return render(
    <PabloThemeProvider>
      <Switch id="custom-id" onChange={onChangeMock} {...props}>
        Hello
      </Switch>
    </PabloThemeProvider>
  );
}
