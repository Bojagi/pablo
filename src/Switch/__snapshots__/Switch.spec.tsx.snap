// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render switch checked 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  box-sizing: border-box;
  position: relative;
  width: calc(2 * 18px + 2 * (clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem) + 1px));
  height: calc(1 * 18px + 2 * (clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem) + 1px));
  border-radius: calc(
      (
          18px + 2 *
            clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem) + 2 *
            1px
        ) * 0.5
    );
  padding: clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem);
  background-color: var(--pbl-theme-colors-common-white);
  border: 1px solid var(--pbl-theme-colors-borders-main);
  opacity: 1;
  -webkit-transition: box-shadow 0.3s ease-in-out;
  transition: box-shadow 0.3s ease-in-out;
}

.emotion-2 {
  box-sizing: border-box;
  width: 18px;
  height: 18px;
  -webkit-transform: translateX(
      18px
    );
  -moz-transform: translateX(
      18px
    );
  -ms-transform: translateX(
      18px
    );
  transform: translateX(
      18px
    );
  border-radius: 50%;
  -webkit-transition: background-color 0.15s ease-in-out,-webkit-transform 0.15s ease-in-out;
  transition: background-color 0.15s ease-in-out,transform 0.15s ease-in-out;
  background-color: var(--pbl-theme-colors-brand-main);
}

.emotion-3 {
  box-sizing: border-box;
  position: absolute;
  display: block;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: pointer;
}

<div>
  <div
    class="emotion-0"
    role="switch"
  >
    <div
      class="emotion-1"
      data-testid="pbl-switch"
    >
      <div
        class="emotion-2"
        data-testid="pbl-switch-handle"
      />
      <input
        checked=""
        class="emotion-3"
        data-testid="pbl-switch-input"
        id="custom-id"
        type="checkbox"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`Render switch unchecked 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  box-sizing: border-box;
  position: relative;
  width: calc(2 * 18px + 2 * (clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem) + 1px));
  height: calc(1 * 18px + 2 * (clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem) + 1px));
  border-radius: calc(
      (
          18px + 2 *
            clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem) + 2 *
            1px
        ) * 0.5
    );
  padding: clamp(0.125rem, 0.10714285714285715rem + 0.08928571428571429vw, 0.1875rem);
  background-color: var(--pbl-theme-colors-common-white);
  border: 1px solid var(--pbl-theme-colors-borders-main);
  opacity: 1;
  -webkit-transition: box-shadow 0.3s ease-in-out;
  transition: box-shadow 0.3s ease-in-out;
}

.emotion-2 {
  box-sizing: border-box;
  width: 18px;
  height: 18px;
  -webkit-transform: translateX(
      0
    );
  -moz-transform: translateX(
      0
    );
  -ms-transform: translateX(
      0
    );
  transform: translateX(
      0
    );
  border-radius: 50%;
  -webkit-transition: background-color 0.15s ease-in-out,-webkit-transform 0.15s ease-in-out;
  transition: background-color 0.15s ease-in-out,transform 0.15s ease-in-out;
  background-color: var(--pbl-theme-colors-gray-500);
}

.emotion-3 {
  box-sizing: border-box;
  position: absolute;
  display: block;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: pointer;
}

<div>
  <div
    class="emotion-0"
    role="switch"
  >
    <div
      class="emotion-1"
      data-testid="pbl-switch"
    >
      <div
        class="emotion-2"
        data-testid="pbl-switch-handle"
      />
      <input
        class="emotion-3"
        data-testid="pbl-switch-input"
        id="custom-id"
        type="checkbox"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`Render switch with label 1`] = `
.emotion-0 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-size: var(--pbl-theme-typography-base-fontSize);
  line-height: var(--pbl-theme-typography-base-lineHeight);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  margin-bottom: var(--pbl-theme-typography-base-marginBottom);
  font-style: normal;
  margin: 0;
  font-family: var(--pbl-theme-typography-body-fontFamily);
  font-size: var(--pbl-theme-typography-body-fontSize);
  line-height: var(--pbl-theme-typography-body-lineHeight);
  font-weight: var(--pbl-theme-typography-body-fontWeight);
  margin-bottom: var(--pbl-theme-typography-body-marginBottom);
  margin-bottom: 0;
  margin-left: clamp(0.5rem, 0.4285714285714286rem + 0.35714285714285715vw, 0.75rem);
}

<label
  class="emotion-0"
  data-testid="pbl-switch-label"
  for="custom-id"
>
  I am a 
  <strong>
    label
  </strong>
</label>
`;
